x-pack/metricbeat/module/azure/app_insights/data.go

```diff
@@ -139,43 +147,156 @@ func EventsMapping(metricValues insights.ListMetricsResultsItem, applicationId s
 			events = append(events, event)
 		}
 	}
-	for _, val := range segValues {
-		for _, seg := range val.Segments {
-			lastSeg := getValue(seg)
-			for _, ls := range lastSeg {
-				events = append(events, createSegEvent(val, ls, applicationId, namespace))
-			}
+
+	groupedByDimensions := groupMetricsByDimension(mValues)
+
+	for _, group := range groupedByDimensions {
+		groupedByTime := groupMetricsByTime(group)
+
+		for ts, group := range groupedByTime {
+			events = append(
+				events,
+				createGroupEvent(group, ts, applicationId, namespace),
+			)
```

Comment:
Why are we grouping by dimensions and later by time instead of grouping by dimension+time like we did in the GCP metrics?